### PR TITLE
Core: Move SplitPane layout from PanelEdit.

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
This PR moves the SplitPane logic from the PanelEditor to its own component. The main reason for this is to reuse the same layout in the upcoming alert definition page (wireframes in #27703). 

**Which issue(s) this PR fixes**:

Part of #27703

**Special notes for your reviewer**:
There's few areas that's very much up for discussion :sweat_smile: 

* Naming  - Not overly attached to `ThreePaneSplit` but not sure what else to go with either? `SplitPaneComponent`, `SplitLayout`?
* Scope - Is the scope too narrow? I've only had the alert definition page and panel edit in mind right now.

### CODE REVIEW

File: public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx [hunk 1]
```diff
@@ -383,7 +300,13 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
       <div className={styles.wrapper} aria-label={selectors.components.PanelEditor.General.content}>
         {this.editorToolbar(styles)}
         <div className={styles.verticalSplitPanesWrapper}>
-          {uiState.isPanelOptionsVisible ? this.renderWithOptionsPane(styles) : this.renderHorizontalSplit(styles)}
+          <ThreePaneSplit
+            horizontalSplitComponents={this.renderPanelAndEditor(styles)}
+            rightPaneComponent={this.renderOptionsPane()}
+            uiState={uiState}
+            updateUiState={updatePanelEditorUIState}
+            rightPanelVisible={uiState.isPanelOptionsVisible}
```
Comment (torkelo) [inline public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx:308]:
```suggestion
            rightPaneVisible={uiState.isPanelOptionsVisible}
```

property should be rightPaneVisible

File: public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+enum Pane {
+  Right,
+  Top,
+}
+
+interface Props {
+  horizontalSplitComponents: React.ReactNode[];
+  rightPaneComponent: React.ReactNode;
+  uiState: { topPaneSize: number | string; rightPaneSize: number | string };
```
Comment (torkelo) [inline public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx:16]: looks like we always treat it as a number, when is it string?

**Review by torkelo — APPROVED** [review]
Think this looks good, will be really nice to reuse all this!

File: public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx (hunk 2 above)
Comment (peterholmberg) [reply public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx:16]: It takes the topPaneSize and rightPaneSize from this model  https://github.com/grafana/grafana/blob/7d8cb6869a2d3cb2bf3dd54dac2b59e60babc975/public/app/features/dashboard/components/PanelEditor/state/reducers.ts#L20

Comment (peterholmberg) [reply public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx:16]: but then we don't handle that case in the component :thinking:

**Pull request merged**